state.go

```diff
@@ -1063,24 +1389,33 @@ func (usr *UpdateStatusReportState) Handle(ctx *StateContext, c Controller) (Sta
 }
 
 type UpdateStatusReportRetryState struct {
+	baseState
 	WaitState
 	reportState  State
-	update       client.UpdateResponse
+	update       datastore.UpdateInfo
 	status       string
 	triesSending int
 }
 
 func NewUpdateStatusReportRetryState(reportState State,
-	update client.UpdateResponse, status string, tries int) State {
+	update *datastore.UpdateInfo, status string, tries int) State {
 	return &UpdateStatusReportRetryState{
-		WaitState:    NewWaitState(MenderStatusReportRetryState, ToNone),
+		baseState: baseState{
+			id: datastore.MenderStatusReportRetryState,
+			t:  ToNone,
+		},
+		WaitState:    NewWaitState(datastore.MenderStatusReportRetryState, ToNone),
 		reportState:  reportState,
-		update:       update,
+		update:       *update,
 		status:       status,
 		triesSending: tries,
 	}
 }
 
+func (usr *UpdateStatusReportRetryState) Cancel() bool {
+	return usr.WaitState.Cancel()
+}
+
 // try to send failed report at lest 3 times or keep trying every
 // 'retryPollInterval' for the duration of two 'updatePollInterval'
 func maxSendingAttempts(upi, rpi time.Duration, minRetries int) int {
```

Comment:
Seems to me that the comment and the code are one refactor iteration off.

Comment: should say "try to send at least minRetries times
Code: `if max <= minRetries`

Reply:
Additionally, shouldn't the code below calculate the max as `2 * upi / rpi` (as the doc claims) before comparing to `minRetries`?

Reply:
Yes, I didn't spend too much time on this function, since it was already used. I fixed the comparison.

As for the second point, I think that is open to interpretation depending on how you read the comment. I think I prefer not to change it now, since it's already used elsewhere, and worst case is that we do a few more retries.